### PR TITLE
[NOCSL] Address tracking endpoint issues

### PR DESCRIPTION
**Pull request by stanlp1**
Some of the tracking functions were sending parameters in the wrong places.

### CODE REVIEW

File: library/src/main/java/io/constructor/core/ConstructorIo.kt [hunk 1]
```diff
@@ -989,8 +989,6 @@ object ConstructorIo {
 
     internal fun trackBrowseResultClickInternal(filterName: String, filterValue: String, customerId: String, variationId: String? = null, resultPositionOnPage: Int, sectionName: String? = null, resultID: String? = null): Completable {
         preferenceHelper.getSessionId(sessionIncrementHandler)
-        val encodedParams: ArrayList<Pair<String, String>> = arrayListOf()
-        resultID?.let { encodedParams.add(Constants.QueryConstants.RESULT_ID.urlEncode() to it.urlEncode()) }
```
[the remark this answers is not in the document]
Comment (stanlp1) [reply library/src/main/java/io/constructor/core/ConstructorIo.kt:993]: result_id isn't actually an accepted url parameter. 


Screen Shot 2022-10-20 at 3 10 05 PM

[the remark this answers is not in the document]
Comment (jjl014) [reply library/src/main/java/io/constructor/core/ConstructorIo.kt:993]: Let's add result id to the post body like we're doing in the recommendation result click and add a test around it

**Review by jjl014 — COMMENTED** [review]
Nice, looking good to me.

Left 1 comment.

**Review by jjl014 — APPROVED** [review]
🚀

**Pull request merged**